demo/riskmodels/fraud_detection/spark_lgbm.py

```diff
@@ -0,0 +1,204 @@
+        dataset = spark.read.format("csv")\
+                            .option("header",  True)\
+                            .option("inferSchema",  True)\
+                            .load(data_path)  
+        return dataset 
+    train_dataset = read_data(train_path)
+    test_dataset = read_data(test_path)
+                              
+    print('train dataset sample:')
+    train_dataset.show(20, False)
+    print('test dataset sample:')
+    test_dataset.show(20, False)
+    return train_dataset, test_dataset
+
+def get_feature_and_label_cols(dataset, **params):
+    label_col = params['label_col']
+    feature_cols = [x for x in dataset.columns if x not in [label_col]]
+    return feature_cols, label_col
+
+def get_vectorassembler(dataset, feature_cols, features='features', label='label'):
+    featurizer = VectorAssembler(
+        inputCols = feature_cols,
+        outputCol = 'features',
+        handleInvalid = 'skip'
+    )
+    dataset = featurizer.transform(dataset)[label, features]
+    return dataset
+
+def train(spark, train_dataset, label_col, **model_params):
+    print('Debug -- model params:\n', model_params)
+    from synapse.ml.lightgbm import LightGBMClassifier
+    model = LightGBMClassifier(isProvideTrainingMetric=True, 
+                               featuresCol='features', labelCol=label_col, 
+                               isUnbalance=True, 
+                               **model_params)
+    model = model.fit(train_dataset)
+    return model
+
+def evaluate(spark, test_result, label_col):
+    evaluator = BinaryClassificationEvaluator(labelCol=label_col, metricName="areaUnderROC")
+    auc = evaluator.evaluate(test_result)
+
+    label = np.array(predictions.select(label_col).collect()).reshape(-1).astype(np.float32)
+    prediction = np.array(predictions.select('probability').collect())[:, :, 1].reshape(-1)
+    ks = ks_2samp(label, prediction)
+    return auc, ks
+
+def write_dataset_to_s3(eval_dataset, eval_out_path, **kwargs):
+    start = time.time()
+    eval_dataset.write.parquet(eval_out_path, mode="overwrite")
+    print('Debug -- write_dataset_to_s3 cost time:', time.time() - start)
+
+def convert_model(lgbm_model: LGBMClassifier or Booster, input_size: int) -> bytes:
+    initial_types = [("input", FloatTensorType([-1, input_size]))]
+    onnx_model = convert_lightgbm(lgbm_model, initial_types=initial_types, target_opset = 9)
+    return onnx_model   
+
+def get_onnx_model(model, len_data_columns):
+    booster_model_str = model.getLightGBMBooster()
+    booster_model_str = booster_model_str.modelStr()
+    booster_model_str = booster_model_str.get()
+    booster = lgb.Booster(model_str = booster_model_str)
+    onnx_model = convert_model(booster, len_data_columns)
+    return onnx_model
+
+def save_onnx_model(onnx_model, model_onnx_path, **kwargs):
+    onnxmltools.utils.save_model(onnx_model,'lightgbm.onnx')
+    loaded_model = onnxmltools.utils.load_model('lightgbm.onnx')
```

Comment:
It's better to take load logic out of save_onnx_model func.

Reply:
fixed